src/nodeProvider.ts

```diff
@@ -38,21 +40,23 @@ export class DidactNodeProvider implements vscode.TreeDataProvider<TreeNode> {
 	}
 
 	// get the list of children for the node provider
-	public getChildren(element?: TreeNode): TreeNode[] {
-        if (!element) {
-            return this.treeNodes;
-        } else {
-            // assume it's a category
-            if (element.label && !(element instanceof TutorialNode)) {
+	public async getChildren(element?: any): Promise<any[]> {
+		if (!element) {
+			return this.treeNodes;
+		} else if (element instanceof TutorialNode) {
+			return await this.processHeadingsForTutorial(element.uri);
+		} else {
+			// assume it's a category
+			if (element.label && !(element instanceof TutorialNode)) {
 				const tutorialCategory = retrieveTreeItemName(element);
-                return this.processTutorialsForCategory(tutorialCategory);
-            }   
-        }
-        return [];
+				return await this.processTutorialsForCategory(tutorialCategory);
+			}   
+		}
+		return [];
 	}
 
 	// add a child to the list of nodes
-	public addChild(newNode: TreeNode, disableRefresh = false, oldNodes: TreeNode[] = this.treeNodes ): TreeNode[] {
+	public addChild(newNode: any, disableRefresh = false, oldNodes: any[] = this.treeNodes ): any[] {
```

Comment:
>     * when there are several headings with same name, the tree displayed a single of them:

At this time, that's due to the fact that headings do not get unique IDs so we are forced to do a text search for them. We search for the first one since there's no way to know which one they may have meant.

Reply:
As for the Any and losing type safety, there are some problems with the current implementation. I would like to address those in a follow-up PR to rearrange the tree class structure a bit to simplify things. Right now there is a mix of vscode.TreeItem and TreeNode implementations as children, which makes this a bit clunky. Let's move it to https://issues.redhat.com/browse/FUSETOOLS2-1050